src/main/java/picard/arrays/illumina/CompareGtcFiles.java

```diff
@@ -0,0 +1,236 @@
+        int length1 = Array.getLength(arrayOne);
+        int length2 = Array.getLength(arrayTwo);
+
+        int diffCount = 0;
+        if (length1 != length2) {
+            compareErrors.add(String.format("%s do not match. Arrays of different lengths. ( %d vs %d )",
+                    type, length1, length2));
+        } else {
+            for (int i = 0; i < length1; i++) {
+                //for floats only compare 3 decimal places
+                if (arrayOne.getClass().getComponentType() == float.class) {
+                    Float float1 = (float) Array.get(arrayOne, i);
+                    Float float2 = (float) Array.get(arrayTwo, i);
+                    if (float1.equals(Float.NaN) || float2.equals(Float.NaN)) {
+                        if (!float1.equals(float2)) diffCount++;
+                    } else {
+                        BigDecimal decimal1 = BigDecimal.valueOf(float1).setScale(3, BigDecimal.ROUND_DOWN);
+                        BigDecimal decimal2 = BigDecimal.valueOf(float2).setScale(3, BigDecimal.ROUND_DOWN);
+                        if (!decimal1.equals(decimal2)) {
+                            diffCount++;
+                        }
+                    }
+                } else if (!Array.get(arrayOne, i).equals(Array.get(arrayTwo, i))) {
+                    diffCount++;
+                }
+            }
+        }
+        return diffCount;
+    }
+
+
+    private void compareArrayOfArrays(Object arrayOfArraysOne, Object arrayOfArraysTwo, String type) {
+        List<String> compareErrors = new ArrayList<>();
+
+        if (checkNulls(arrayOfArraysOne, arrayOfArraysTwo, type)) return;
+
+        int differences = 0;
+        int length1 = Array.getLength(arrayOfArraysOne);
+        int length2 = Array.getLength(arrayOfArraysTwo);
+        if (length1 != length2) {
+            compareErrors.add(String.format("%s do not match. Arrays of different lengths. ( %d vs %d )",
+                    type, length1, length2));
+        } else {
+            //iterate over the first array
+            for (int i = 0; i < length1; i++) {
+                //iterate over the second array
+                Object innerArrayOne = Array.get(arrayOfArraysOne, i);
+                Object innerArrayTwo = Array.get(arrayOfArraysTwo, i);
+                differences += arrayDifferences(innerArrayOne, innerArrayTwo, type, compareErrors);
+            }
+            if (differences > 0) {
+                compareErrors.add(String.format("%s do not match. %d elements of the array differ.", type, differences));
+            }
+        }
+        checkErrors(type, compareErrors);
+    }
+
+    private void checkErrors(String type, List<String> compareErrors) {
+        if (compareErrors.size() > 0) {
+            errors.addAll(compareErrors);
+        } else {
+            log.info(type + " IDENTICAL");
+        }
+    }
+
+    private boolean checkNulls(Object arrayOne, Object arrayTwo, String type) {
```

Comment:
These parameters are not necessarily arrays. Maybe `Object objectOne, Object objectTwo` would be better
```suggestion
    private boolean checkNulls(Object objectOne, Object objectTwo, String type) {
```

Reply:
renamed.